### ISSUE TITLE
Why wav2vec2-ctc beam search decoder use logits as AM score?

### ISSUE THREAD

**Issue by SeunghyunSEO** [facebookresearch/fairseq#4317]
First of all, i want to say thank you for sharing this amazing pretrained model and decoder :)
I have a simple question about beam search decoder [here](https://github.com/pytorch/fairseq/blob/main/examples/speech_recognition/new/decoders/base_decoder.py#L48), especially below line

```python
encoder_out = model(**encoder_input)
if hasattr(model, "get_logits"):
    emissions = model.get_logits(encoder_out)
else:
    emissions = model.get_normalized_probs(encoder_out, log_probs=True)
return emissions.transpose(0, 1).float().cpu().contiguous()
```

As you can see, the decoder use am score as logits not log probability. (because w2v2-ctc has `get_logits` function)

To my best knowledge, beam search decoder have to use log probability of am and lm to compute beam score.

<img width="403" alt="스크린샷 2022-03-29 오후 11 52 56" src="https://user-images.githubusercontent.com/48202736/160640344-af3dae5c-4ddc-46fa-91a0-0f1870084e88.png">

(equation is from END-TO-END ASR: FROM SUPERVISED TO SEMI-SUPERVISED LEARNING WITH MODERN ARCHITECTURES)

Nevertheless, i could reproduce the paper scores.
but I have just wondering why fairseq implemented CTC beam search decoder using ctc logit scores.

As far as I check, actual score about certain audio input was like

```python
'''
(Pdb) nbest_results[0].amScore
21540.401293754578
(Pdb) nbest_results[0].lmScore
-258.7231324315071
'''
```

I would very much appreciate if you answer this question ! (👀 @alexeib )
